todo_app/app.py

```diff
@@ -16,19 +17,58 @@
 def index():
 
     cardList = get_items()
-    
-        
+     
     return render_template('index.html', cardList=cardList)    
     
 
 @app.route('/add', methods = ['POST'])
 def add_Task():
     
-    # get the title data from the Add Task form field in index.html
-    # add it to the list
-    newTask = request.form.get('newTask')
-    add_item(newTask)
+    title = request.form.get('title')
+    description = request.form.get('description')
+    list = request.form.get('listName')
+    listId = get_list_by_name(list)
+    
+    add_item(title, description, listId) 
     
-    # send user back to starting app route
     return redirect('/')
-    
\ No newline at end of file
+
+
+@app.route('/task/', methods =['GET'])
+def get_Task():
+
+    taskId = request.args.get('taskId')
+    taskName = request.args.get('taskName')
+    taskDescription = request.args.get('taskDescription')
+    taskListName =  request.args.get('taskListName')
+
+    return render_template('task.html', taskId=taskId, taskName=taskName, taskListName=taskListName, taskDescription=taskDescription)
+
+
+@app.route('/update/', methods=['GET','PUT'])
```

Comment:
We'd usually action this with a PUT (or a POST if we're stuck using HTML Forms!) though to keep it simple using links I don't think a GET is unreasonable - but even if you're choosing a non-standard method I'd still advise locking it down to the smallest set of methods you can, just to prevent potentially confusing behaviour (like if this is expecting a POST, then will users pass the info through query args, or will they use the body of a request?)